### PR TITLE
refactor(replace nats): add blockdevice proto

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**
Signed-off-by: Abhinandan Purkait <abhinandan.purkait@mayadata.io>

### CODE REVIEW

**Review by tiagolobocastro — APPROVED** [review]

File: control-plane/grpc/proto/v1/blockdevice/blockdevice.proto [hunk 1]
```diff
@@ -0,0 +1,77 @@
+syntax = "proto3";
+
+import "v1/misc/common.proto";
+
+package v1.blockdevice;
+
+message BlockDevice {
+  // entry in /dev associated with device
+  string devname = 1;
+  // currently "disk" or "partition"
```
Comment (paulyoong) [inline control-plane/grpc/proto/v1/blockdevice/blockdevice.proto:10]: Nit: `currently` -> `current`?

File: control-plane/grpc/proto/v1/blockdevice/blockdevice.proto [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  string devtype = 2;
+  // major device number
+  uint32 devmajor = 3;
+  // minor device number
+  uint32 devminor = 4;
+  // device model - useful for identifying mayastor devices
+  string model = 5;
+  // official device path
+  string devpath = 6;
+  // list of udev generated symlinks by which device may be identified
```
Comment (paulyoong) [inline control-plane/grpc/proto/v1/blockdevice/blockdevice.proto:20]: Nice comment :+1:

File: control-plane/grpc/proto/v1/blockdevice/blockdevice.proto [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+message GetBlockDevicesRequest {
+  // id of the mayastor instance
+  string node_id = 1;
+  // specifies whether to get all devices or only usable devices
+  bool all = 2;
+}
+
+// Reponse to the GetNodes request
```
Comment (paulyoong) [inline control-plane/grpc/proto/v1/blockdevice/blockdevice.proto:71]: Typo: `GetNodes` -> `GetBlockDevices`?

**Review by paulyoong — APPROVED** [review]

File: control-plane/grpc/proto/v1/blockdevice/blockdevice.proto (hunk 1 above)
Comment (Abhinandan-Purkait) [reply control-plane/grpc/proto/v1/blockdevice/blockdevice.proto]: Done.

File: control-plane/grpc/proto/v1/blockdevice/blockdevice.proto (hunk 3 above)
Comment (Abhinandan-Purkait) [reply control-plane/grpc/proto/v1/blockdevice/blockdevice.proto]: Done.

**Comment by Abhinandan-Purkait** [comment]
bors merge

File: control-plane/grpc/proto/v1/blockdevice/blockdevice.proto (hunk 1 above)
Comment (tiagolobocastro) [reply control-plane/grpc/proto/v1/blockdevice/blockdevice.proto]: Actually I think the previous was probably correct. It was saying currently the devtype is either "disk" or "partition"

**Pull request merged**